cluster-autoscaler/cloudprovider/gce/gce_manager.go

```diff
@@ -206,22 +206,19 @@ func (m *gceManagerImpl) registerMig(mig Mig) bool {
 
 // GetMigSize gets MIG size.
 func (m *gceManagerImpl) GetMigSize(mig Mig) (int64, error) {
-	if migSize, found := m.cache.GetMigTargetSize(mig.GceRef()); found {
-		return migSize, nil
-	}
-	targetSize, err := m.GceService.FetchMigTargetSize(mig.GceRef())
-	if err != nil {
-		return -1, err
-	}
-	m.cache.SetMigTargetSize(mig.GceRef(), targetSize)
-	return targetSize, nil
+	return m.cache.GetMigTargetSize(mig.GceRef())
```

Comment:
I don't really like moving this logic inside the cache itself. Sth called cache.GetMigTargetSize doesn't imply in any way that the cache will self-refresh, or that the negative result isn't just a cache miss (but that the actual check was performed). I'd rather keep the cache simple, and use it as cache.

Reply:
Moved out of cache